sql/wsrep_mysqld.cc

```diff
@@ -3151,3 +3151,27 @@ bool wsrep_consistency_check(THD *thd)
 {
   return thd->wsrep_consistency_check == CONSISTENCY_CHECK_RUNNING;
 }
+
+void wsrep_wait_ready(THD *thd)
+{
```

Comment:
there used to be a function for such wait: wsrep_ready_wait(), but it has been refactored out at some point. However, the declaration remains still in wsrep_mysqld.h (looked in 10.6 sources). This ready wait used cond signal, and was introduced with commit: df4dd593f29a

Reply:
We still seem to have that condition variable in 10.4 but I could not find it used really anywhere . Do you suggest that I will start using it?

Reply:
the old code is better removed as obsolete. However, the ready wait loop is using here sleep() and old implementation relied on cond signal to arrive, which is better for resource usage. If cond signal delivery is reliable, consider using that method instead of sleep loop.